### DISCUSSION TITLE
No serverless functions working on deployed application

### DISCUSSION THREAD

**Discussion by zreecespieces** [vercel/community#3778] [Help] [answered]
### Summary

None of my serverless functions are working after I deployed my project to Vercel. All of them work just fine locally and even running locally with `vercel dev` in the CLI. However, on the deployed application every single call to a serverless function times out and fails. Not a single one works not even the most basic example.

### Example

https://chrisramirezart.com/api/testRoute2

### Steps to Reproduce

 For example, https://chrisramirezart.com/api/testRoute2, is set to the default API route from the documentation:

```js
export default function handler(req, res) {
  res.status(200).json({ message: 'Hello from Next.js!' })
}
```

In the logs I can see it is failing with a 504 with no message (request ID: `fwbjk-1693033262041-fa54a4934e18`) when I try to go to the function directly at `https://chrisramirezart.com/api/testRoute2`. However, if I make the request from within my application code to that same route I get the message: `{"error": {"code": "504", "message": "An error occurred with your deployment"}}`
Netlify App URL: `chris-ramirez-9rxw0k1i1-zacharydreece.vercel.app`
Associated Custom Domain: `chrisramirezart.com`

Running locally with `npm run dev` works just fine and running locally with `vercel dev` it also works just fine:
`http://localhost:3000/api/testRoute2` -> {"message":"Hello from Next.js!"}

Any help would be greatly appreciated. I'm on the Hobby plan if that's relevant but for this basic example of a simple static response that shouldn't really be a factor unless I'm unaware of something. 

Here is my package.json:

```json
{
  "name": "nextjs",
  "version": "5.0.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "post-update": "echo \"codesandbox preview only, need an update\" && yarn upgrade --latest"
  },
  "dependencies": {
    "@emotion/cache": "latest",
    "@emotion/react": "latest",
    "@emotion/server": "latest",
    "@emotion/styled": "latest",
    "@formspark/use-formspark": "^0.1.39",
    "@mailchimp/mailchimp_marketing": "^3.0.78",
    "@mui/icons-material": "^5.10.2",
    "@mui/material": "latest",
    "@react-hookz/web": "^15.1.0",
    "@react-spring/web": "^9.5.5",
    "luxon": "^3.4.0",
    "next": "latest",
    "node-fetch": "2.6.1",
    "notistack": "^2.0.5",
    "prop-types": "latest",
    "react": "latest",
    "react-dom": "latest",
    "react-stickynode": "^4.1.0"
  },
  "devDependencies": {
    "eslint": "latest",
    "eslint-config-next": "latest"
  }
}
```

From yarn.lock:
```
next@latest:
  version "12.2.5"
```

Failed Response Example:
```
504: GATEWAY_TIMEOUT
Code: FUNCTION_INVOCATION_TIMEOUT
ID: cle1::fwbjk-1693033262041-fa54a4934e18
```

**Comment by amyegan** [comment, marked as answer]
Hey @zreecespieces. Do your serverless functions need any tokens or environment variable values to connect to external services? If env vars are needed, please double check that they were added in project settings. Connection issues from missing tokens are a common reason for [504: FUNCTION_INVOCATION_TIMEOUT](https://vercel.com/docs/errors/application-error-codes#504:-function_invocation_timeout) errors.

Another common cause of 504 errors is when a function needs to run for a long time and exceeds the [execution timeout threshold](https://vercel.com/docs/limits/overview#serverless-function-execution-timeout). The limit is currently 10 seconds on a personal Hobby account or 60 seconds for Pro teams. If you need a longer execution time, [Edge Functions](https://vercel.com/docs/functions/edge-functions) require an [initial response](https://vercel.com/docs/functions/edge-functions/limitations#maximum-initial-response-time) within 30 seconds and allow you to [continue streaming responses](https://vercel.com/docs/functions/edge-functions/streaming) after that.

**Comment by zreecespieces** [comment]
Updating to the latest version of next resolved the issue. All environment variables were working correctly and the timeout wasn't an issue because even a dummy API route that returned a static string was timing out as well. Must have been some issue with that version of next + deployment pipeline/environment
